### ISSUE TITLE
Action with namespace: TypeError: Cannot redefine property

### ISSUE THREAD

**Issue by clarkdo** [ktsn/vuex-class#9]
When use Action with namespace, page will throw "TypeError: Cannot redefine property: ..."

Part of codes like blow:

```javascript
import Vue from 'vue'
import Component, {Getter, Action, namespace } from 'class-component'
const DemoGetter = namespace('demo', Getter)
const DemoAction = namespace('demo', Action)

@Component
export default class Demo extends Vue {
  @DemoAction selectCity
  @Getter authUser
  @DemoGetter city
}
```

According to my research, after vuex-class defined the actions on vm, vue "initMethods" will define all methods onto vm.

```javascript
function initMethods (vm, methods) {
  var props = vm.$options.props;
  console.log('methods: ' +JSON.stringify( methods))
  for (var key in methods) {
    vm[key] = methods[key] == null ? noop : bind(methods[key], vm);
    if (process.env.NODE_ENV !== 'production') {
      if (methods[key] == null) {
        warn(
          "method \"" + key + "\" has an undefined value in the component definition. " +
          "Did you reference the function correctly?",
          vm
        );
      }
      if (props && hasOwn(props, key)) {
        warn(
          ("method \"" + key + "\" has already been defined as a prop."),
          vm
        );
      }
    }
  }
}
```

And then vue-class-component method '_init' may redefine them again, so the error occurred.

```javascript
Component.prototype._init = function () {
        var _this = this;
        var keys = Object.getOwnPropertyNames(vm);
        if (vm.$options.props) {
            for (var key in vm.$options.props) {
                if (!vm.hasOwnProperty(key)) {
                    keys.push(key);
                }
            }
        }
        keys.forEach(function (key) {
            if (key.charAt(0) !== '_') {
                Object.defineProperty(_this, key, {
                    get: function () { return vm[key]; },
                    set: function (value) { return vm[key] = value; }
                });
            }
        });
    };
```

Does this have any solution or pr?

**Comment by kota65535** [comment]
Is there any plan to fix it?

**Comment by Eazymov** [comment]
`@DemoAction selectCity() {}` // works fine

I hope there is a better solution

**Comment by dimamarksman** [comment]
The same error. 
Actions do not work for me at all, no matter with namespace or without.

**Comment by ktsn** [comment]
Could you provide self-contained reproduction via jsfidde, codepen, github repo or so on?

**Comment by dimamarksman** [comment]
@ktsn I created repo https://github.com/dimamarksman/vue-vuex-class-test
Please take a look.

**Comment by ktsn** [comment]
Fixed via vue-class-component v6.1.0

**Comment by dimamarksman** [comment]
@ktsn Thank you!
